openmetadata-ui/src/main/resources/ui/src/components/Modals/EntityNameModal/EntityNameModal.component.tsx

```diff
@@ -65,7 +61,9 @@ const EntityNameModal: React.FC<Props> = ({
       open={visible}
       title={
         <Typography.Text strong data-testid="header">
-          {t('label.edit-glossary-name')}
+          {t('label.edit-entity', {
+            entity: allowRename ? t('label.name') : t('label.display-name'),
```

Comment:
It will be good to have header as props instead making a logic here.
Something to update in future